homeassistant/components/avea/__init__.py

```diff
@@ -1 +1,40 @@
-"""The avea component."""
+"""The avea integration."""
+import asyncio
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS = ["light"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Avea component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Avea from a config entry."""
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
```

Comment:
Here you are removing the hass.data[DOMAIN][entry.entry_id] object, but I cant see you actually creating this object anywhere. This code looks identical to the code I copied for my integration so I suspect you may have copied this part and don't require it.

Reply:
You are absolutely right, the code is left over from the configflow generation

Fixed: https://github.com/home-assistant/core/pull/42102/commits/4cf5bcc4b54257eafe6de1da616b631445bbb95e